metricbeat/module/jolokia/jmx/config.go

```diff
@@ -0,0 +1,71 @@
+package jmx
+
+import "strings"
+
+type MetricSetup struct {
+	MBean      string
+	Attributes []Attribute
+}
+
+type Attribute struct {
+	Attr  string
+	Field string
+}
+
+type MetricSetConfig struct {
+	Body        string
+	Mapping     map[string]string
+	Application string
+	Instance    string
+}
+
+type SliceSet map[string][]string
+
+func (s SliceSet) Add(key, value string) {
+	_, ok := s[key]
+	if !ok {
+		s[key] = make([]string, 0, 30)
+	}
+	s[key] = append(s[key], value)
+}
+
+//parse MetricSetConfigRaw to MetricSetConfig
+func parseConfig(mapping []MetricSetup, app string, instance string) (MetricSetConfig, error) {
+
+	currBody, mapBuilt := buildRequestBodyAndMapping(mapping)
+	config := MetricSetConfig{
+		currBody, mapBuilt, app, instance,
+	}
+	debugf("Jolokia request config will be included: %#v", config)
+
+	return config, nil
+}
+
+func buildRequestBodyAndMapping(mapping []MetricSetup) (string, map[string]string) {
+	var requestBodyStructure = make(SliceSet)
+	var responseMapping = make(map[string]string)
+	for _, metricSetup := range mapping {
+		for _, attribute := range metricSetup.Attributes {
+			requestBodyStructure.Add(metricSetup.MBean, attribute.Attr)
+			responseMapping[metricSetup.MBean+"_"+attribute.Attr] = attribute.Field
+		}
+	}
+	return marshalJSONRequest(requestBodyStructure), responseMapping
+}
+
+func marshalJSONRequest(this SliceSet) string {
```

Comment:
No specific reason to use custom marshaling here, this was just a quick-and-dirty solution to test how it would work that eventually was left behind.